src/tools/opensc-explorer.c

```diff
@@ -1670,28 +1763,31 @@ static int do_get_data(int argc, char **argv)
 static int do_put_data(int argc, char **argv)
 {
 	unsigned int tag;
-	u8 buf[8192];
-	size_t buflen = sizeof(buf);
+	u8 buf[SC_MAX_EXT_APDU_BUFFER_SIZE];
+	size_t i, len;
 	int r;
 
-	if (argc != 2)
+	if (argc < 2)
 		return usage(do_put_data);
 
 	/* Extract DO's tag */
 	tag = strtoul(argv[0], NULL, 16);
 
 	/* Extract the new content */
-	/* buflen is the max length of reception buffer */
-	r = parse_string_or_hexdata(argv[1], buf, &buflen);
-	if (r < 0) {
-		printf("unable to parse data\n");
-		return -1;
+	for (i = 1, len = 0; i < (unsigned) argc; i++)   {
+		size_t len0 = sizeof(buf) - len;
+
+		if ((r = parse_string_or_hexdata(argv[i], buf + len, &len0)) < 0) {
```

Comment:
Isn't it counter intuitive to have muliple arguments? You're effectively ommiting the white space between two arguments when writing to the card.

Reply:
Maybe. But ...
If you call it with only one argument, there is **no** change in behaviour to the previous version.

This is because the command line is split into white-space (i.e. `" \t\n"`) separated tokens before the the functions are parsed and the handler for the individual function is called. This is how opensc-explorer works.
With this setup, it is (and this is unchanged!)
* impossible after the split to know what combination of white-space existed between the arguments before the split
* only possible to include white-space as a part of a hex-string (e.g. `20:08`) or a double-quoted string (e.g. `" "`).

My change only treats `do_put` the same as `apdu` by accepting multiple arguments.

Perhaps a clarifying sentence in the doc could help.
Let me see ...